src/Makefile.am

```diff
@@ -273,24 +273,53 @@ sptps_speed_SOURCES += \
 endif
 endif
 
+tinc_SOURCES = \
+	$(tinc_common_SOURCES) \
+	tincctl.c tincctl.h
+
+tincd_SOURCES = \
+	$(tincd_common_SOURCES) \
+	process.c process.h \
+	tincd.c
+
+LIBS = @LIBS@ -lm $(CODE_COVERAGE_LIBS)
+tinc_LDADD = @LIBS@ $(READLINE_LIBS) $(CURSES_LIBS)
+tincd_LDADD = @LIBS@ $(MINIUPNPC_LIBS)
+
 if MINIUPNPC
-tincd_SOURCES += upnp.h upnp.c
-tincd_LDADD = $(MINIUPNPC_LIBS)
+tincd_common_SOURCES += upnp.h upnp.c
 tincd_LDFLAGS = -pthread
 endif
 
 if MINGW
 sptps_test_LDFLAGS = -pthread
 endif
 
-tinc_LDADD = $(READLINE_LIBS) $(CURSES_LIBS)
 sptps_speed_LDADD = -lrt
 
-LIBS = @LIBS@ -lm $(CODE_COVERAGE_LIBS)
-
 if TUNEMU
 LIBS += -lpcap
 endif
 
 AM_CFLAGS = -DCONFDIR=\"$(sysconfdir)\" -DRUNSTATEDIR=\"$(runstatedir)\" -DLOCALSTATEDIR=\"$(localstatedir)\" -DSBINDIR=\"$(sbindir)\" -iquote. $(CODE_COVERAGE_CFLAGS)
 AM_CPPFLAGS = $(CODE_COVERAGE_CPPFLAGS)
+
+if HAVE_CMOCKA
+## Use TAP log driver (gives each test result separately in `make check` output)
+LOG_DRIVER = env AM_TAP_AWK='$(AWK)' $(SHELL) $(top_srcdir)/tap-driver.sh
+
+AM_TESTS_ENVIRONMENT = \
+	export CMOCKA_MESSAGE_OUTPUT=TAP;
+
+AM_CFLAGS += -iquote unittest
+
+tinc_test_LDADD = $(tinc_LDADD) -lcmocka
```

Comment:
This causes a warning from automake that there is no binary `tinc_test`. Changing the `_LDADD` suffix to `_ldadd` fixes it.